pkg/client/patch.go

```diff
@@ -148,7 +164,16 @@ func MergeFromWithOptions(obj runtime.Object, opts ...MergeFromOption) Patch {
 	for _, opt := range opts {
 		opt.ApplyToMergeFrom(options)
 	}
-	return &mergeFromPatch{from: obj, opts: *options}
+	return &mergeFromPatch{patchType: types.MergePatchType, createPatch: createMergePatch, from: obj, opts: *options}
+}
+
+// StrategicMergeFrom creates a Patch that patches using the strategic-merge-patch strategy with the given object as base.
+func StrategicMergeFrom(obj runtime.Object, opts ...MergeFromOption) Patch {
```

Comment:
Can you please add godocs here and to `MergeFrom` that explain how StrategicMergePatch and MergePatch differ and when to use which?

Reply:
Sure, good idea. Done.
WDYT now?